virtual/activation_cache.go

```diff
@@ -83,16 +84,22 @@ func (a *activationsCache) ensureActivation(
 	moduleID,
 	actorID string,
 
-	blacklistedServerID string,
+	replicasNumber int,
+	blacklistedServerIDs []string,
 ) ([]types.ActorReference, error) {
 	// Ensure we have a short timeout when communicating with registry.
 	ctx, cc := context.WithTimeout(ctx, defaultActivationCacheTimeout)
 	defer cc()
 
+	isServerIdBlacklisted := make(map[string]bool)
+	for _, s := range blacklistedServerIDs {
+		isServerIdBlacklisted[s] = true
+	}
+
 	if a.c == nil {
 		// Cache disabled, load directly.
 		return a.ensureActivationAndUpdateCache(
-			ctx, namespace, moduleID, actorID, nil, blacklistedServerID)
+			ctx, namespace, moduleID, actorID, replicasNumber, nil, isServerIdBlacklisted, blacklistedServerIDs)
```

Comment:
isServerIdBlacklisted and blacklistedServerIDs are redundant. However, for efficient usage of blacklisting we need to have a map, but at the same time, for calling EnsureActivation on the registry, we need a list (due to json format)